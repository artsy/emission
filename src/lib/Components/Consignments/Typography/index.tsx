import * as React from "react"
import { StyleSheet, Text, TextProperties, TextStyle } from "react-native"

import colors from "../../../../data/colors"
import fonts from "../../../../data/fonts"

const LargeHeadline = (props: TextProperties) => {
  const children: string = (props as any).children
  const style = [styles.largeDefault, props.style || {}, styles.largeRequired]
  return <Text key={children} style={style}>{children}</Text>
}

const SmallHeadline = (props: TextProperties) => {
  const children: string = (props as any).children
  const style = [styles.smallDefault, props.style || {}, styles.smallRequired]
  return <Text key={children} style={style}>{children}</Text>
}

const Subtitle = (props: TextProperties) => {
  const children: string = (props as any).children
  const style = [styles.subtitleDefault, props.style || {}, styles.subtitleRequired]
  return <Text key={children} style={style}>{children}</Text>
}

const BodyText = (props: TextProperties) => {
  const children: string = (props as any).children
  const style = [styles.bodyDefault, props.style || {}, styles.bodyRequired]
  return <Text key={children} style={style}>{children}</Text>
}

export { LargeHeadline, SmallHeadline, Subtitle, BodyText }

interface Styles {
  largeRequired: TextStyle
  largeDefault: TextStyle
  smallRequired: TextStyle
  smallDefault: TextStyle
  subtitleRequired: TextStyle
  subtitleDefault: TextStyle
  bodyRequired: TextStyle
  bodyDefault: TextStyle
}

const styles = StyleSheet.create<Styles>({
  largeDefault: {
    fontSize: 30,
    color: "white",
    textAlign: "center",
    paddingLeft: 20,
    paddingRight: 20,
  },

  largeRequired: {
    fontFamily: fonts["garamond-regular"],
  },

  smallDefault: {
    fontSize: 30,
    color: "white",
    textAlign: "center",
  },

  smallRequired: {
    fontFamily: fonts["garamond-regular"],
  },

  subtitleDefault: {
    fontSize: 20,
    color: colors["gray-medium"],
    paddingLeft: 25,
    paddingRight: 25,
  },

  subtitleRequired: {
    fontFamily: fonts["garamond-italic"],
  },

  bodyDefault: {
    fontSize: 20,
    color: colors["gray-medium"],
    paddingLeft: 25,
    paddingRight: 25,
    marginTop: 18,
    marginBottom: 18,
    textAlign: "center",
  },

  bodyRequired: {
    fontFamily: fonts["garamond-regular"],
  },
})
