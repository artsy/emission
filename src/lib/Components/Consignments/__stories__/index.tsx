import { storiesOf } from "@storybook/react-native"
import * as React from "react"
import { View } from "react-native"

import "./Consignments.story"
import "./Style.story"

import * as bottomAlignedButton from "./BottomAligned.story"
import * as search from "./Search.story"
import * as todo from "./Todo.story"

// Just a quick interface so you know the API
export interface AutoStory {
  name: string
  allStates: any[]
  component: any
}
// Converts a set of state and a component type into a set of stories
const autoStories: AutoStory[] = [search, bottomAlignedButton, todo]

export const Wrapper = props =>
  <View style={{ flex: 1, backgroundColor: "black", padding: 20, marginTop: 60 }}>
    {props.children}
  </View>

// Allows stories to strictly be about their state and the
// component itself.

autoStories.forEach(storybook => {
  const stories = storiesOf(storybook.name)

  storybook.allStates.forEach(element => {
    const name = Object.keys(element)[0]
    stories.add(name, () =>
      <Wrapper>
        <storybook.component {...element[name]} />
      </Wrapper>
    )
  })
})
