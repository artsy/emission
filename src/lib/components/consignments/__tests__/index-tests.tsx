import * as React from "react"
import "react-native"
import * as renderer from "react-test-renderer"
import Consignments from "../"

it("Sets up the right view hierarchy", () => {
    const tree = renderer.create(<Consignments />).toJSON()
    expect(tree).toMatchSnapshot()
})
