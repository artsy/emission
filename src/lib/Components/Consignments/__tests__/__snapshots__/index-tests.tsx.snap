// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sets up the right view hierarchy 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTNavigator
    interactivePopGestureEnabled={undefined}
    onNavigationComplete={[Function]}
    requestedTopOfStack={0}
    style={
      Object {
        "flex": 1,
      }
    }
    vertical={undefined}
  >
    <RCTNavItem
      initialRoute={
        Object {
          "component": [Function],
          "title": "Welcome",
        }
      }
      navigationBarHidden={true}
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          undefined,
          undefined,
        ]
      }
      title="Welcome"
      translucent={true}
    >
      <RCTScrollView
        style={
          Object {
            "backgroundColor": "black",
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "paddingTop": 40,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 30,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                    "textAlign": "center",
                  },
                  Object {},
                  Object {
                    "fontFamily": "AGaramondPro-Regular",
                  },
                ]
              }
            >
              Sell works from your collection through our partner network
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "marginTop": 20,
                  "width": 300,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderColor": "white",
                    "borderRadius": 40,
                    "borderWidth": 2,
                    "height": 80,
                    "justifyContent": "center",
                    "width": 80,
                  }
                }
              >
                <Image
                  source={1}
                />
              </View>
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#cccccc",
                      "fontSize": 20,
                      "marginBottom": 18,
                      "marginTop": 18,
                      "paddingLeft": 25,
                      "paddingRight": 25,
                      "textAlign": "center",
                    },
                    Object {},
                    Object {
                      "fontFamily": "AGaramondPro-Regular",
                    },
                  ]
                }
              >
                Sell work from your collection through our partner network.
              </Text>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderColor": "white",
                    "borderRadius": 40,
                    "borderWidth": 2,
                    "height": 80,
                    "justifyContent": "center",
                    "width": 80,
                  }
                }
              >
                <Image
                  source={1}
                />
              </View>
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#cccccc",
                      "fontSize": 20,
                      "marginBottom": 18,
                      "marginTop": 18,
                      "paddingLeft": 25,
                      "paddingRight": 25,
                      "textAlign": "center",
                    },
                    Object {},
                    Object {
                      "fontFamily": "AGaramondPro-Regular",
                    },
                  ]
                }
              >
                Get your work placed in an upcoming sale.
              </Text>
            </View>
            <View
              style={
                Object {
                  "height": 43,
                  "marginTop": 20,
                  "width": 320,
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "rgba(255, 255, 255, 1)",
                      "borderColor": "#e5e5e5",
                      "borderWidth": 1,
                      "flex": 1,
                      "justifyContent": "center",
                    },
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "fontSize": 12,
                        },
                        Object {
                          "color": "black",
                          "fontFamily": "AGaramondPro-Regular",
                          "opacity": 1,
                        },
                        Object {
                          "fontFamily": "Avant Garde Gothic ITCW01Dm",
                        },
                      ]
                    }
                  >
                    GET STARTED
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </RCTNavItem>
  </RCTNavigator>
</View>
`;
